src/psyclone/psyir/symbols/routinesymbol.py

```diff
@@ -0,0 +1,47 @@
+# -----------------------------------------------------------------------------
+# BSD 3-Clause License
+#
+# Copyright (c) 2020, Science and Technology Facilities Council.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+# FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+# COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+# ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Authors R. W. Ford and S. Siso, STFC Daresbury Lab
+# -----------------------------------------------------------------------------
+
+''' This module contains the RoutineSymbol.'''
+
+from __future__ import absolute_import
+from psyclone.psyir.symbols.symbol import Symbol
+
+
+class RoutineSymbol(Symbol):
+    '''Symbol identifying a callable routine.'''
+
+    def __str__(self):
+        return "routine(name='{0}')".format(self.name)
```

Comment:
Can you make this consistent with the other Symbol `__str__`, which is `self.name [: <PROPERTIES>]`. I know this is not the best representation but when we do `symbol_table.view()` it gives a more tabular representation. In this case it has no properties yet.

Maybe at this point we should add the Symbol type as part of the textual representation to all symbols `self.name : Routine[<PROPERTIES>]` ? Although we can start with just this one as chaning all will need to touch a lot of tests.

Reply:
OK. I've used the class name i.e. `RoutineSymbol`, rather than just `Routine` and I've not changed the other symbol strings (thanks for not making me do this :-) ). If you don't like me using the class name then let me know and I'll change it. There are currently no properties so I've omitted the `<>`, however I guess we might want to add linked/not linked at some point (as we do with ContainerSymbol).